embassy-stm32/src/crc/v1.rs

```diff
@@ -0,0 +1,55 @@
+use crate::pac::CRC as PAC_CRC;
+use crate::peripherals::CRC;
+use crate::rcc::sealed::RccPeripheral;
+
+pub struct Crc {
+    _peripheral: CRC,
+}
+
+impl Crc {
+    /// Instantiates the CRC32 peripheral and initializes it to default values.
+    pub fn new(peripheral: CRC) -> Self {
```

Comment:
Most peripherals take an Unborrow, i.e. Unborrow<Target = CRC> so the HAL can be used with both &mut and owned types. Would it make sense to allow that here too?

Reply:
I am unfamiliar with the semantics on this, though armed with @Dirbaio 's comment below about `reclaim` not being part of the API I guess it makes sense.